src/res_scanner.ml

```diff
@@ -54,28 +54,28 @@ let position scanner = Lexing.{
 }
 
 let next scanner =
-  if scanner.rdOffset < (Bytes.length scanner.src) then (
+  if scanner.rdOffset < Bytes.length scanner.src then (
     scanner.offset <- scanner.rdOffset;
     let ch = (Bytes.get [@doesNotRaise]) scanner.src scanner.rdOffset in
     scanner.rdOffset <- scanner.rdOffset + 1;
-    scanner.ch <- int_of_char ch
+    scanner.ch <- ch
   ) else (
     scanner.offset <- Bytes.length scanner.src;
-    scanner.ch <- -1
+    scanner.ch <- CharacterCodes.eof
```

Comment:
I've tried removing this hacky encoding by using a `isEndOfFile` field. But too many scanner helpers in this file rely on implicitly matching on some characters and sending the end-of-file `ch` case to an else branch. So using `isEndOfFile` is causing some infinite loops. Reverted back to hacking -1 into char.

We're not supposed to do this, but this isn't worst than what it is already.